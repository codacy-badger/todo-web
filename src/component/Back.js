import React from 'react';
import { withRouter } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronLeft } from '@fortawesome/free-solid-svg-icons';
import 'component/Back.sass';

function Back ({ history }) {

  const goBack = () => {
    history.goBack();
  }
  return (
    <FontAwesomeIcon id="back-component" onClick={() => goBack()} icon={ faChevronLeft } />
  )
}

export default withRouter(Back);
