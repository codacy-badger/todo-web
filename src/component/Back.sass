#back-component
  opacity: 1
  cursor: pointer
  margin-right: 30px