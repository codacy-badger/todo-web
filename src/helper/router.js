import { history } from "router";

export function redirect(path = "/") {
  history.push(path);
}
