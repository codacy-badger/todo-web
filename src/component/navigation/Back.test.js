import React from "react";
import renderer from "react-test-renderer";
import { Router, Switch } from "react-router-dom";
import { history } from "router";
import Back from "component/navigation/Back";

describe("Back component", () => {
  test("should render back", () => {
    const component = renderer
      .create(
        <Router history={history}>
          <Switch>
            <Back />
          </Switch>
        </Router>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
