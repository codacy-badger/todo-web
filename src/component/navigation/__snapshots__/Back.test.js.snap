// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back component should render back 1`] = `
<span>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-chevron-left fa-w-10 "
    data-for="tooltip-back"
    data-icon="chevron-left"
    data-prefix="fas"
    data-tip="Go Back"
    focusable="false"
    id="back-component"
    onClick={[Function]}
    role="img"
    style={Object {}}
    viewBox="0 0 320 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <div
    className="__react_component_tooltip place-bottom type-dark "
    data-id="tooltip"
    id="tooltip-back"
  />
</span>
`;
